pyClient/zeth/joinsplit.py

```diff
@@ -316,30 +394,114 @@ class ZethClient:
     """
     def __init__(
             self,
+            web3: Any,
             prover_client: ProverClient,
+            mk_tree_depth: int,
             mixer_instance: Any,
+            merkle_root: str,
             zksnark: IZKSnarkProvider):
         self._prover_client = prover_client
-        self._mixer_instance = mixer_instance
+        self.web3 = web3
         self._zksnark = zksnark
+        self.mixer_instance = mixer_instance
+        self.mk_tree_depth = mk_tree_depth
+        self.merkle_root = merkle_root
+
+    @staticmethod
+    def open(
+            web3: Any,
+            prover_client: ProverClient,
+            mk_tree_depth: int,
+            mixer_instance: Any,
+            zksnark: IZKSnarkProvider) -> ZethClient:
+        """
+        Create a client for an existing Zeth deployment.
+        """
+        return ZethClient(
+            web3,
+            prover_client,
+            mk_tree_depth,
+            mixer_instance,
+            contracts.get_merkle_root(mixer_instance).hex(),
+            zksnark)
+
+    @staticmethod
+    def deploy(
+            web3: Any,
+            prover_client: ProverClient,
+            mk_tree_depth: int,
+            deployer_eth_address: str,
+            zksnark: IZKSnarkProvider,
+            token_address: Optional[str] = None) -> ZethClient:
+        """
+        Deploy Zeth contracts.
+        """
+        print("[INFO] 1. Fetching verification key from the proving server")
+        vk_obj = prover_client.get_verification_key()
+        vk_json = zksnark.parse_verification_key(vk_obj)
+
+        print("[INFO] 2. Received VK, writing verification key...")
+        write_verification_key(vk_json)
+
+        print("[INFO] 3. VK written, deploying smart contracts...")
+        (proof_verifier_interface, otsig_verifier_interface, mixer_interface) = \
+            contracts.compile_contracts(zksnark)
+        hasher_interface, _ = contracts.compile_util_contracts()
+        (mixer_instance, initial_merkle_root) = contracts.deploy_contracts(
+            web3,
+            mk_tree_depth,
+            proof_verifier_interface,
+            otsig_verifier_interface,
+            mixer_interface,
+            hasher_interface,
+            vk_json,
+            deployer_eth_address,
+            4000000,
```

Comment:
Let's pass the deployment gas as an argument to this function (we can use a default value to `4000000` to keep the tests unaffected)

Reply:
Added a `deploy_gas` argument.